File: src/server/src/main/java/com/project/application/models/workout/WorkoutExerciseDetailsResource.java
package com.project.application.models.workout;

import com.project.domain.workout.Exercise;
import com.project.domain.workout.WorkoutExercise;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;
import java.util.UUID;

@Serdeable
public record WorkoutExerciseDetailsResource(UUID id, List<SetResource> sets) {
    public static WorkoutExerciseDetailsResource from(WorkoutExercise exercise) {
        return new WorkoutExerciseDetailsResource(exercise.getId(), exercise.getSets().stream().map(SetResource::from)
                .toList());
    }
}

File: src/server/src/main/kotlin/com/project/application/models/workout/WorkoutExerciseDetailsResource.kt
package com.project.application.models.workout

import com.project.domain.workout.WorkoutExercise
import com.project.domain.workout.WorkoutSet
import io.micronaut.serde.annotation.Serdeable
import java.util.UUID

@Serdeable
data class WorkoutExerciseDetailsResource(val id: UUID, val exerciseId: UUID?, val sets: List<WorkoutSetResource>) {

    companion object {
        @JvmStatic
        fun from(exercise: WorkoutExercise): WorkoutExerciseDetailsResource {
            return WorkoutExerciseDetailsResource(exercise.id, exercise.exerciseId, exercise.sets.map { set: WorkoutSet -> WorkoutSetResource.from(set) })
        }
    }
}
